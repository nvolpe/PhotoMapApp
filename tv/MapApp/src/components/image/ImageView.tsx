import React from 'react';
import {StyleSheet, Image, View} from 'react-native';
import CameraRoll from '@react-native-community/cameraroll';
import {Colors} from 'react-native/Libraries/NewAppScreen';

interface Props {
  photo: CameraRoll.PhotoIdentifier | null;
}

const ImageView = (props: Props) => {
  const {photo} = props;

  const renderImage = () => {
    const imageSize = 500;
    const imageStyle = [styles.image, {width: imageSize, height: imageSize}];
    if (photo) {
      return (
        <Image
          testID={'image'}
          source={require('../../testData/testImages/test1.jpg')}
          style={imageStyle}
        />
      );
    } else {
      // if we have a missing image.. show it here?
      return <View testID={'emptyView'} />;
    }
  };

  return <View style={styles.container}>{renderImage()}</View>;
};

export default ImageView;

const styles = StyleSheet.create({
  container: {
    width: '100%',
    height: '100%',
  },
  sectionTitle: {
    fontSize: 24,
    fontWeight: '600',
    color: Colors.black,
  },
  fakeImageStyle: {
    width: 500,
    height: 500,
    backgroundColor: Colors.black,
  },
  image: {
    margin: 4,
  },
});
