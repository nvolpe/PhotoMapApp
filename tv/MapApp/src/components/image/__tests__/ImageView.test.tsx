import React from 'react';
import {render, fireEvent} from '@testing-library/react-native';
import ImageView from '../ImageView';
import CameraRoll from '@react-native-community/cameraroll';

// what is fetch mock?
// https://github.com/MattAgn/NewsletterApp/blob/master/src/complex-version/pages/Home/__tests__/Home.test.tsx
//https://github.com/MattAgn/NewsletterApp/blob/master/src/complex-version/pages/Home/test-solutions/Home.testSolution.tsx

// it('should display succesful message on successful subscribe', async () => {
//   // Setup
//   fetchMock.post(NEWSLETTER_ENDPOINT, 200);
//   const page = renderPage(<Home />);
//   // What the user sees
//   const EmailInput = page.getByPlaceholder(wording.emailPlaceholder);
//   const ValidateButton = page.getByText(wording.validateEmail);
//   // What the user does
//   fireEvent.changeText(EmailInput, 'hello@bam.co');
//   fireEvent.press(ValidateButton);
//   // What feedback the user should expect
//   const SuccessMessage = await waitForElement(() =>
//     page.queryByText(wording.subscriptionSuccessful),
//   );
//   expect(SuccessMessage).toBeTruthy();
// });

test('Image view renders correctly with a valid image', () => {
  const testImage: CameraRoll.PhotoIdentifier = {
    node: {
      timestamp: 1605397864,
      group_name: 'All Photos',
      type: 'image',
      location: {
        speed: 0,
        latitude: 32.849425000000004,
        longitude: -117.27258055555555,
        heading: 0,
        altitude: 4.7,
      },
      image: {
        playableDuration: null,
        uri: 'xxx.jpg',
        filename: 'PXL_20201114_235103605.jpg',
        width: 4032,
        height: 3024,
        fileSize: 3619392,
      },
    },
  };

  const componentUnderTest = render(<ImageView photo={testImage} />);
  expect(componentUnderTest.getByTestId('image')).toBeTruthy();

  expect(componentUnderTest.toJSON()).toMatchSnapshot();
});

test('Image view does not render an image view without a photo', () => {
  const testImage: CameraRoll.PhotoIdentifier | null = null;
  const componentUnderTest = render(<ImageView photo={testImage} />);
  expect(componentUnderTest.getByTestId('emptyView')).toBeTruthy();
  expect(componentUnderTest.toJSON()).toMatchSnapshot();
});
