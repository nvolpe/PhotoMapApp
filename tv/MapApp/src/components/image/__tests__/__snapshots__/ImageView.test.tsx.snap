// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image view does not render an image view without a photo 1`] = `
<View
  style={
    Object {
      "height": "100%",
      "width": "100%",
    }
  }
>
  <View
    testID="emptyView"
  />
</View>
`;

exports[`Image view renders correctly with a valid image 1`] = `
<View
  style={
    Object {
      "height": "100%",
      "width": "100%",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/testData/testImages/test1.jpg",
      }
    }
    style={
      Array [
        Object {
          "margin": 4,
        },
        Object {
          "height": 500,
          "width": 500,
        },
      ]
    }
    testID="image"
  />
</View>
`;
